### DISCUSSION TITLE
How to pass multiple commands to tic80 --cli using --cmd=<str>

### DISCUSSION THREAD

**Discussion by stephen010x** [nesbox/TIC-80#2608] [Q&A] [answered]
I am creating a simple build system for myself using *luacc* and *windows batch* in order to create my game code using multiple lua files (with my choice of external editor)

The goal here is to create a batch file (or makefile) that can build and run my project without changing any of the sprites or mapping, etc.

The idea would be to first run luacc to merge all of the project files into a single `output.lua`
And then to soon after run this sequence of commands in `tic80.exe --cli`
```
load game.lua
import code output.lua
save game.lua
```
This way if there were any changes I made to game.lua through the native sprite, sound, or map editors, they would remain unaffected.

And even if there is a way to do everything I want with a single command, I figure it would be rather useful to have a way to run multiple commands through `tic80 --cli`

Basically, to summarize my question, I just want a way to **automatically run multiple commands through tic80's command line**.

How might I do this? 
Does such a feature exist? 
If not, can we add it?

Also, thank you to anyone who is willing to spare their time on my question/request.

**Comment by stephen010x** [comment, marked as answer]
Gentlemen, it would appear that not only 30 minutes later, I managed to answer my own question.

In my attempts to figure out how to do this, I was looking through the source code of [TQ-Bundler](https://github.com/scambier/TQ-Bundler) by @scambier, a project that aims to help in the development of tic80 games using multiple code files. (though too feature lacking for my use)
In his code I found that he uses *single ampersand* `&` separators between every command in his `--cmd` string.

Here is an example of the usage using the Windows command prompt:
```cmd
C:\Windows\System32>tic80.exe --cli --cmd="help version & help tri"

1.1.2837 Pro
---=== API ===---
tri(x1 y1 x2 y2 x3 y3 color)

This function draws a triangle filled with color, using the supplied vertices.
```

Is this documented somewhere in the wiki that I just missed?

**Reply by scambier** [reply]
(out of curiosity, which features do you think are lacking?)

**Reply by stephen010x** [reply]
I know that a single ampersand is kind of obvious. The reason I didn't figure this out on my own is that my go-to command separator is the double ampersand `&&` rather than the single ampersand `&`. And for some reason it never occurred to me to try them both.

**Reply by scambier** [reply]
I was talking about TQ-Bundler, you summoned me ^^

**Reply by stephen010x** [reply]
@scambier
I meant no disrespect. It is a wonderful tool.
And I apologize for the summon. I didn't realize using @ would notify you. This was my first time using the @ for github, and it was only my intention to link to your project and your profile to give credence for my discovery.

Originally I was going to use your tool for my project. The only reason I hadn't is that I wanted to run tic80 with certain flags to test the program, or to just not run the program at all and just build it. Neither were things I could figure out how to do with your tool, so I instead opted for my own makefile.

**Reply by Skeptim** [reply]
> Is this documented somewhere in the wiki that I just missed?

Yes, there is a wiki page about command line arguments with example using `--cmd` with  `&` https://github.com/nesbox/TIC-80/wiki/Command-line-arguments#examples

I am open to suggestions to improve the wiki. Is it the wiki page that is hidden or did you miss the examples on the page?

**Reply by scambier** [reply]
> it was only my intention to link to your project and your profile to give credence for my discovery.

No worry, that's how I took it, so thanks!

I simply asked out of curiosity, but yes, passing custom arguments isn't supported :) Though if you don't use `--tic`, TQ-Bundler should just build your game without launching TIC-80.

Good luck with your own makefile 👍
